### PR TITLE
[Broker Interceptor] Fix Pulsar didn't respond error messages when throw InterceptException

### LINKED ISSUE #11649: [Broker Interceptor] Pulsar didn't respond error messages when throw InterceptException
**Describe the bug**
Cluster didn't respond error messages when hitting the resource limit

**To Reproduce**
Set the limit of number of tenants to 4, create a new tenant when there're already 4 tenants.

**Expected behavior**
Respond valid error messages.

**Screenshots**
> curl -v -XPUT -H "Authorization: Bearer $TOKEN" https://xxx.com/admin/v2/tenants/abc -d '{"allowedClusters": ["free"],"adminRoles":["admin"]}'
> *   Trying 198.18.8.55...
> * TCP_NODELAY set
> * Connected to xxx.com (x.x.x.x) port 443 (#0)
> * ALPN, offering h2
> * ALPN, offering http/1.1
> * successfully set certificate verify locations:
> *   CAfile: /etc/ssl/cert.pem
>   CApath: none
> * TLSv1.2 (OUT), TLS handshake, Client hello (1):
> * TLSv1.2 (IN), TLS handshake, Server hello (2):
> * TLSv1.2 (IN), TLS handshake, Certificate (11):
> * TLSv1.2 (IN), TLS handshake, Server key exchange (12):
> * TLSv1.2 (IN), TLS handshake, Request CERT (13):
> * TLSv1.2 (IN), TLS handshake, Server finished (14):
> * TLSv1.2 (OUT), TLS handshake, Certificate (11):
> * TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
> * TLSv1.2 (OUT), TLS change cipher, Change cipher spec (1):
> * TLSv1.2 (OUT), TLS handshake, Finished (20):
> * TLSv1.2 (IN), TLS change cipher, Change cipher spec (1):
> * TLSv1.2 (IN), TLS handshake, Finished (20):
> * SSL connection using TLSv1.2 / ECDHE-RSA-AES256-SHA384
> * ALPN, server did not agree to a protocol
> * Server certificate:
> *  subject: CN=xxx.com
> *  start date: Nov 10 07:21:31 2020 GMT
> *  expire date: Feb  8 07:21:31 2021 GMT
> *  subjectAltName: host "xxx.com" matched cert's "xxx.com"
> *  issuer: C=US; O=Let's Encrypt; CN=Let's Encrypt Authority X3
> *  SSL certificate verify ok.
> > PUT /admin/v2/tenants/abc HTTP/1.1
> > Host: xxx.com
> > User-Agent: curl/7.64.1
> > Accept: */*
> > Authorization: Bearer
> > Content-Length: 52
> > Content-Type: application/x-www-form-urlencoded
> >
> * upload completely sent off: 52 out of 52 bytes
> < HTTP/1.1 401 Unauthorized
> < Content-Length: 0
> < Server: Jetty(9.4.29.v20200521)
> <
> * Connection #0 to host xxxx left intact
> * Closing connection 0

**Additional context**

### PR DESCRIPTION
**Pull request by Technoboy-**
Fixes #11649

### Documentation

This is a bug fix, and do not need to update doc.

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/web/WebService.java [hunk 1]
```diff
@@ -159,11 +159,12 @@ public void addServlet(String path, ServletHolder servletHolder, boolean require
                 context.setAttribute(key, value);
             });
         }
+        ExceptionHandler handler = new ExceptionHandler();
```
Comment (eolivelli) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/web/WebService.java:162]: we can push this "new" inside the "if" block, this way we can save us from creating it if not needed

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/web/ExceptionHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    @Test
+    @SneakyThrows
+    public void testHandle() {
+        ExceptionHandler handler = new ExceptionHandler();
+        HttpServletResponse response = Mockito.mock(HttpServletResponse.class);
+        handler.handle(response, new InterceptException(HttpStatus.PRECONDITION_FAILED_412, "Reach the max tenants [5] restriction"));
+        Mockito.verify(response).sendError(Mockito.anyInt(), Mockito.anyString());
+        handler.handle(response, new InterceptException(HttpStatus.INTERNAL_SERVER_ERROR_500, "internal exception"));
+        Mockito.verify(response, Mockito.times(2)).sendError(Mockito.anyInt(), Mockito.anyString());
```
Comment (eolivelli) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/web/ExceptionHandlerTest.java:45]: can we match the exact error code and message ?

**Review by eolivelli — CHANGES_REQUESTED** [review]
overall LGTM

I left a few minor comments

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/web/WebService.java (hunk 1 above)
Comment (Technoboy-) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/web/WebService.java:162]: So quick review, many thanks to you ,Enrico.

Comment (Technoboy-) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/web/WebService.java:162]: I will update soon

**Review by codelipenghui — APPROVED** [review]

**Review by 315157973 — APPROVED** [review]

**Review by eolivelli — APPROVED** [review]
Lgtm

**Pull request merged**